Parser/pegen/pegen.c

```diff
@@ -2160,3 +2160,94 @@ _PyPegen_nonparen_genexp_in_call(Parser *p, expr_ty args)
         "Generator expression must be parenthesized"
     );
 }
+
+OperatorTermPair *
+_PyPegen_operator_term_pair(Parser *p, Token* oper, expr_ty factor)
+{
+    OperatorTermPair *a = PyArena_Malloc(p->arena, sizeof(OperatorTermPair));
+    if (!a) {
+        return NULL;
+    }
+    a->oper = oper;
+    a->factor = factor;
+    return a;
+}
+
+static operator_ty
+_PyPegen_get_binary_operator(const Token *t)
+{
+    switch (t->type) {
+        case VBAR:
+            return BitOr;
+        case CIRCUMFLEX:
+            return BitXor;
+        case AMPER:
+            return BitAnd;
+        case LEFTSHIFT:
+            return LShift;
+        case RIGHTSHIFT:
+            return RShift;
+        case PLUS:
+            return Add;
+        case MINUS:
+            return Sub;
+        case STAR:
+            return Mult;
+        case AT:
```

Comment:
We can't support the feature_version check for matmul. Honestly I think we could just drop that check, but there's a test for it.

Reply:
I have passed down the parser struct and we raise now for feature version.